File: articles/iteration-recursion-and-control-flow/java/src/main/java/org/contentcatalyst/algorithms/ControlFlowAudit.java
package org.contentcatalyst.algorithms;
public class ControlFlowAudit {
  static int factorial(int n){ return n <= 0 ? 1 : n * factorial(n - 1); }
  public static void main(String[] args){
    System.out.println("test_name,value");
    System.out.println("factorial_5," + factorial(5));
    System.out.println("iterative_sum," + (1 + 2 + 3));
  }
}

File: articles/iteration-recursion-and-control-flow/cpp/control_flow_audit.cpp
#include <iostream>
int factorial(int n){ return n <= 0 ? 1 : n * factorial(n-1); }
int main(){ std::cout << "test_name,value\nfactorial_5," << factorial(5) << "\niterative_sum," << (1+2+3) << "\n"; }
